cmd/legacy/legacy.go

```diff
@@ -166,21 +166,24 @@ func (l LegacyFileSet) GetBlock(ts int64) (*hashmap.Map, error) {
 			logrus.StandardLogger().Warnf("source / destination IP v4 / v6 mismatch: %s / %s, will convert to IPv6\n", sip, dip)
 		}
 
-		var K types.Key
 		var V types.Counters
 
-		if sip.Is4() && dip.Is4() {
-			K = types.NewV4KeyStatic(sip.As4(), dip.As4(), dportBlock[i*2:i*2+2], protoBlock[i])
-		} else {
-			K = types.NewV6KeyStatic(sip.As16(), dip.As16(), dportBlock[i*2:i*2+2], protoBlock[i])
-		}
-
 		V.BytesRcvd = binary.BigEndian.Uint64(bytesRcvdBlock[i*8 : i*8+8])
 		V.BytesSent = binary.BigEndian.Uint64(bytesSentBlock[i*8 : i*8+8])
 		V.PacketsRcvd = binary.BigEndian.Uint64(pktsRcvdBlock[i*8 : i*8+8])
 		V.PacketsSent = binary.BigEndian.Uint64(pktsSentBlock[i*8 : i*8+8])
 
-		data.SetOrUpdate(K, V.BytesRcvd, V.BytesSent, V.PacketsRcvd, V.PacketsSent)
+		if sip.Is4() && dip.Is4() {
+			data.V4Map.SetOrUpdate(
+				types.NewV4KeyStatic(sip.As4(), dip.As4(), dportBlock[i*2:i*2+2], protoBlock[i]),
+				V.BytesRcvd, V.BytesSent, V.PacketsRcvd, V.PacketsSent,
+			)
+		} else {
```

Comment:
I don't know if this case can ever happen in practice: but what if only `sip` or `dip` is IPv4 and the other is IPv6. The check would be more exact if we do `} else if sip.Is6() && dip.Is6() { .. } else { return fmt.Errorf("holy sh*t") }`

Reply:
The thing is: It _does_ happen, and it's "ok-ish" (because of the way the standard library treats e.g. `0.0.0.0`, see a few lines further above:
```
if sip.Is4() != dip.Is4() && !sip.IsUnspecified() {
	logrus.StandardLogger().Warnf("source / destination IP v4 / v6 mismatch: %s / %s, will convert to IPv6\n", sip, dip)
}
```
But you're right, in theory `IsUnspecified()` should cover these cases. I'll slightly extend this warning logic to throw a specific warning if we have a mix of IPv4 and IPv6 and it's _not_ unspecified (and skip that entry). Either we'll never see that warning or we get a hint at potential remaining edge cases...